src/Codeception/Module/REST.php

```diff
@@ -527,6 +529,67 @@ protected function encodeApplicationJson($method, $parameters)
         return $parameters;
     }
 
+    private function formatFilesArray(array $files)
+    {
+        foreach ($files as $name => $value) {
+            if (is_string($value)) {
+                $this->checkFileBeforeUpload($value);
+
+                $files[$name] = [
+                    'name' => basename($value),
+                    'tmp_name' => $value,
+                    'size' => filesize($value),
+                    'type' => $this->getFileType($value),
+                    'error' => 0,
+                ];
+                continue;
+            } elseif (is_array($value)) {
+                if (isset($value['tmp_name'])) {
+                    $this->checkFileBeforeUpload($value['tmp_name']);
+                    if (!isset($value['name'])) {
+                        $value['name'] = basename($value);
```

Comment:
If `$value` is an array, why call `basename($value)` on it? This leads to the error `basename() expects parameter 1 to be string, array given`.